tests/ImageKit/Upload/UploadTest.php

```diff
@@ -60,144 +67,330 @@ private function stubHttpClient($response)
+        $this->checkFormData($stream,$boundary,"overwriteAITags","false");
+        $this->checkFormData($stream,$boundary,"overwriteCustomMetadata","true");
+        $this->checkFormData($stream,$boundary,"customMetadata",json_encode($fileOptions['customMetadata']));
+        
+        // Response Check
+        UploadTest::assertEquals(json_encode($this->uploadSuccessResponseObj), json_encode($response->result));
+    }
+
+    /**
+     *
+     */
+    public function testFileUploadMissingUseUniqueFileName()
+    {
+        $fileOptions = [
+            'file'  =>  'http://lorempixel.com/640/480/',
+            'fileName'  =>  'test_file_name',
+            'isPrivateFile' => true
+        ];
+
+        $mockBodyResponse = Utils::streamFor(json_encode($fileOptions));
+
+        $mock = new MockHandler([
+            new Response(200, ['X-Foo' => 'Bar'], $mockBodyResponse)
         ]);
 
-        $response = json_decode(json_encode($res), true);
+        $handlerStack = HandlerStack::create($mock);
 
-        UploadTest::assertEquals([
-            'width' => 1000
-        ], $response['success']);
-    }
+        $container = [];
+        $history = Middleware::history($container);
 
+        $handlerStack->push($history);
+        
+        $mockClient = new ImageKit(
+            'testing_public_key',
+            'testing_private_key',
+            'https://ik.imagekit.io/demo',
+            Transformation::DEFAULT_TRANSFORMATION_POSITION,
+            $handlerStack
+        );
+        
+        $response = $mockClient->uploadFile($fileOptions);
+        
+        $requestBody = $container[0]['request']->getBody();
+        $requestHeaders = $container[0]['request']->getHeaders();
+        $boundary = str_replace("multipart/form-data; boundary=","",$requestHeaders["Content-Type"][0]);
+
+        UploadTest::assertArrayHasKey("Content-Type",$requestHeaders);
+        UploadTest::assertStringStartsWith("multipart/form-data; boundary=",$requestHeaders['Content-Type'][0]);
+
+        $stream = Utils::streamFor($requestBody);
+        $stream = str_replace('\r\n',' ',json_encode($stream->getContents()));
+
+        $this->checkFormData($stream,$boundary,"file",$fileOptions['file']);
+        $this->checkFormData($stream,$boundary,"fileName",$fileOptions['fileName']);
+        $this->checkFormData($stream,$boundary,"isPrivateFile","true");
+        UploadTest::assertStringNotContainsString("useUniqueFileName",$stream);
+    }
 
     /**
-     * @deprecated
+     *
      */
-    public function testDeprecatedFileUploadIfSuccessful()
+    public function testFileUploadMissingIsPrivateFileUseUniqueFileName()
     {
-        $mockBodyResponse = Utils::streamFor(json_encode([
-            'width' => 1000
-        ]));
+        $fileOptions = [
+            'file'  =>  'http://lorempixel.com/640/480/',
+            'fileName'  =>  'test_file_name',
+            "tags" => implode(",",["abd", "def"]),
```

Comment:
Pass `["abd", "def"]` in tags as an array and this test case should continue to pass if SDK is properly converting array to comma seperated string.